lib/elements/satin_column.py

```diff
@@ -359,6 +359,24 @@ def contour_underlay(self):
     def contour_underlay_stitch_length(self):
         return max(self.get_float_param("contour_underlay_stitch_length_mm", 1.5), 0.01)
 
+    @property
+    @param(
+        'contour_underlay_stitch_tolerance_mm',
+        _('Stitch tolerance'),
+        tooltip=_(
+            'All stitches must be within this distance from the path. '
+            'A lower tolerance means stitches will be closer together. '
+            'A higher tolerance means sharp corners may be rounded. '
+            'Defaults to stitch length.'
+        ),
+        unit='mm',
+        group=_('Contour Underlay'),
+        type='float',
+    )
+    def contour_underlay_stitch_tolerance(self):
+        tolerance = self.get_float_param("contour_underlay_stitch_tolerance_mm", self.contour_underlay_stitch_length)
+        return max(tolerance, 0.01)
```

Comment:
This currently chokes on unset tolerances due to the implicit mm to px conversion of `get_float_param` getting applied twice, breaking underlay on all older files. I would suggest just giving tolerance a default of 0.2mm as this gives sensible behaviour for underlay.

Reply:
Thanks for the catch. Wow, I wasn't aware we were inconsistent on returning param values. While we return the default value directly for int values, we do mm calculations on floats. I don't think that's good. We should return the defaults directly as they are given without any further changes.

Reply:
If we're going to do a large refactor making that large a change (as currently all of the hardcoded defaults are in mm), we should just split off `get_mm_param_as_px` into its own function and get rid of the magic param names.

Reply:
I added a fix to my current PR #2830

Reply:
Unfortunately, the running stitch algorithm tends to blow up when tolerances are that large, especially when its input is also such a coarse sample. Using the stitch length as the tolerance worked on exactly none of the curved satin columns I tested it on, all of which worked with the old underlay.

My proposed default value of 0.2mm is not chosen arbitrarily. It is chosen as half of the default inset, which is value needed by the running stitch and sampling algorithm to "just work" and reliably stay underneath the satin column.

Reply:
Basically, the 0.2 default is for backwards compatibility. Larger tolerances might be useful in some cases, but applying them as a default will break existing designs.